lib/js/vdom.js

```diff
@@ -541,6 +542,46 @@ const VDom = (function() {
       return [String(val)];
     } else if (lbl == "HTMLRaw") {
       return [h("span", { innerHTML: String(val) }, [])];
+    } else if (lbl == "HTMLWidget") {
+      // Internal widget setup for the virtual-dom library.
+      function Widget(tagName, attrs) {
+        this.type = "Widget";
+        this.tagName = tagName;
+        this.attrs = attrs;
+      }
+      Widget.prototype.init = function () {
+        const elem = document.createElement(this.tagName);
+        LINKEDLIST.forEach(this.attrs, function(attr) {
+          elem.setAttribute(attr[1], attr[2]);
+        });
+        return elem;
+      }
+
+      // The innards of a widget are *not* affected by diffing.
+      Widget.prototype.update = function (prev, elem) { }
+
+      // Check to see whether an 'id' has been set explicitly.
+      let id = null;
+      LINKEDLIST.forEach(val[2], function(attr) {
+        if (attr[1] == "id") {
+          id = attr[1];
+        }
+      });
+
+      // If the ID has been set, use the ID to persist over renders by
+      // storing in the 'widgets' dictionary.
+      let widget = null;
+      if (id != null) {
```

Comment:
Can `id` be `undefined`? If not, then you should probably use `!==` rather than `!=`.